### PR TITLE
Optimize archive_builds task

### PR DESCRIPTION
**Pull request by stsewd**
A couple of things I noticed:

- We were doing an extra query for each build
  to retrieve the `commands` relationship.
  I have used prefetch_related to fix this.
- We were serializing the whole object,
  but only using the `commands` part.
  I have fixed this by serializing only the commands
  instead, and using `only` in the qs.

Also, now that we are using prefetch_related,
we can't use `iterator()` since that will
ignore prefetch_related.

And since we can't use iterator,
now the whole thing will be loaded into memory,
so I have reduced the limit from 5K to 2K,
I have tested this query on web extra and with 2K the
process isn't killed.

Currently, we have `285648` builds that aren't on storage

### CODE REVIEW

File: readthedocs/builds/tasks.py [hunk 1]
```diff
@@ -205,23 +205,28 @@ def archive_builds_task(days=14, limit=200, include_cold=False, delete=False):
...
+        if commands:
+            for cmd in commands:
                 if len(cmd['output']) > MAX_BUILD_COMMAND_SIZE:
                     cmd['output'] = cmd['output'][-MAX_BUILD_COMMAND_SIZE:]
                     cmd['output'] = "... (truncated) ...\n\nCommand output too long. Truncated to last 1MB.\n\n" + cmd['output']  # noqa
                     log.warning('Truncating build command for build.', build_id=build.id)
-            output = BytesIO()
-            output.write(json.dumps(data).encode('utf8'))
-            output.seek(0)
+            output = BytesIO(json.dumps(commands).encode('utf8'))
```
Comment (humitos) [inline readthedocs/builds/tasks.py:229]: Don't we need to keep the `.seek(0)` here?

File: readthedocs/builds/tasks.py [hunk 2]
```diff
@@ -187,7 +187,7 @@ def run(self, *args, **kwargs):
 
         lock_id = '{0}-lock'.format(self.name)
         days = kwargs.get('days', 14)
-        limit = kwargs.get('limit', 5000)
+        limit = kwargs.get('limit', 2000)
```
Comment (humitos) [inline readthedocs/builds/tasks.py:190]: If we are worried about the query being killed, why we don't use a lower number here and trigger it more frequently instead?

**Review by humitos — APPROVED** [review]
I think this is fine and it's a good initial test to see if the query improves. That said, how are we going to measure that it's working better than before?

File: readthedocs/builds/tasks.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/builds/tasks.py:190]: The task is being executed every 30  minutes currently, I thought that was okay, but yeah, we can do something like 1500 builds each 15 min.

**Comment by stsewd** [comment]
>  how are we going to measure that it's working better than before?

I have executed the query on production, it's faster. But we can check newrelic or AWS, I think Eric was checking the CPU utilization graph on AWS.

File: readthedocs/builds/tasks.py (hunk 2 above)
Comment (humitos) [reply readthedocs/builds/tasks.py:190]: As long as we are archiving builds faster than producing them, it's fine.

**Pull request merged**